lpc5500 family: re-unlock debug access after reset

LeBlue opened the pull request:
After a reset, debug access must be re-unlocked. 

Fixes #1303

I tested this with a LPCxpresso55s28 board (target ```lpc55s28```) and a custom board with ```lpcs5526```.

If needed, I can squash the two commits.

flit reviewed (changes requested):
Only one change requested: please add your copyright to the file's header comment below existing copyrights.

flit commented:
/azp run

flit commented on pyocd/target/family/target_lpc5500.py:
```diff
@@ -234,6 +237,13 @@ def unlock(self, dm_ap):
+        # unlock debug access after reset
+        super(LPC5500Family, self).reset(reset_type)
+
+        self.check_locked_state(None)
```
Just noticed. We probably don't need this. Normally, `SoCTarget` will pass the reset request to its selected core, thus calling `CortexM_LPC5500.reset()`.

https://github.com/pyocd/pyOCD/blob/9b1a6319ef749a2c73d0cc5b341d60aea8c70047/pyocd/core/soc_target.py#L207-L214

LeBlue replied:
yes, it is not necessary. Removing it.

LeBlue commented:
> Only one change requested: please add your copyright to the file's header comment below existing copyrights.
Done

Cleaned up whitespace I added accidentally.

Also removed the added ```halt``` in ```reset_and_halt``` after ```reset```. This was a leftover change from the first iteration.
